usecases/sharding/state.go

```diff
@@ -24,10 +24,11 @@ import (
 const shardNameLength = 12
 
 type State struct {
-	IndexID  string              `json:"indexID"` // for monitoring, reporting purposes. Does not influence the shard-calculations
-	Config   Config              `json:"config"`
-	Physical map[string]Physical `json:"physical"`
-	Virtual  []Virtual           `json:"virtual"`
+	IndexID             string              `json:"indexID"` // for monitoring, reporting purposes. Does not influence the shard-calculations
+	Config              Config              `json:"config"`
+	Physical            map[string]Physical `json:"physical"`
+	Virtual             []Virtual           `json:"virtual"`
+	PartitioningEnabled bool                `json:"partitioningEnabled"`
```

Comment:
`PartitioningEnabled bool ` is not new and so are the rest of State's methods.
I think we need here to rebase on the current master branch

Reply:
It's not new, but I had to make it a public field so it could be marshaled as part of the schema sharding state

Reply:
The rest of the State fields were just reformatted